src/main/kotlin/org/rust/ide/hints/RustParameterInfoHandler.kt

```diff
@@ -0,0 +1,159 @@
+package org.rust.ide.hints
+
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.lang.parameterInfo.*
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.parentOfType
+
+/**
+ * Provides funcions/methods afruments hint.
+ */
+class RustParameterInfoHandler : ParameterInfoHandler<PsiElement, RustArgumentsDescription> {
```

Comment:
When I change `PsiElement` to `RustArgListElement` it just stops displaying hints and doesn't call `updateParameterInfo()`. It looks like there's more behind the scenes than just tying methods together. I checked a couple of implementations in other plugins that I could find quickly ([Dart](https://github.com/JetBrains/intellij-plugins/blob/d0fcca865b56466415f13763cb3c517830c8a708/Dart/src/com/jetbrains/lang/dart/ide/info/DartParameterInfoHandler.java) and [ColdFusion](https://github.com/JetBrains/intellij-plugins/blob/d0fcca865b56466415f13763cb3c517830c8a708/CFML/src/com/intellij/coldFusion/UI/editorActions/CfmlParameterInfoHandler.java)), they use `PsiElement` as well.

But if you're sure it must work with `RustArgListElement`, I'll have a look under the hood a bit later.